build.assets/changelog.sh

```diff
@@ -1,15 +1,204 @@
+#
+#     gh pr edit --body-file /tmp/changelog
+#
+
+# Set by check_prereq - either jq or gojq
+JQ=
+
+main() {
+	set -eu
+
+	check_prereq || exit 1
+
+	local branch last_version
+	branch=$(get_branch) || exit 1
+	last_version=$(get_last_version) || exit 1
+
+	since=$(git show -s --date=format:'%Y-%m-%dT%H:%M:%S%z' --format=%cd "${last_version}") || {
+		echo 'Cant get timestamp of last release' >&2
+		return 1
+	}
+	since_e=$(git -C e show -s --date=format:'%Y-%m-%dT%H:%M:%S%z' --format=%cd "${last_version}") || {
+		echo 'Cant get enterprise timestamp of last release' >&2
+		return 1
+	}
+	e_time=$(git log -n 1 --date=format:'%Y-%m-%dT%H:%M:%S%z' --format=%cd e) || {
+		echo 'Cant get last modified time of "e" ref' >&2
+		return 1
+	}
+
+	list_prs "${branch}" "${since}"
+	printf '\nEnterprise:\n'
+	(cd e && list_prs "${branch}" "${since_e}" "${e_time}")
+
+	return 0
+}
+
+check_prereq() {
+	if command -v jq >/dev/null 2>&1; then
+		JQ=jq
+		return 0
+	fi
+	if command -v gojq >/dev/null 2>&1; then
+		JQ=gojq
+		return 0
+	fi
+
+	echo 'jq or gojq not installed. Install gojq easily with:' >&2
+	echo 'go install github.com/itchyny/gojq/cmd/gojq@latest' >&2
+	return 1
+}
+
+get_branch() {
+	# If BASE_BRANCH is set, just use that. Otherwise try to figure it out.
+	if [[ -n "${BASE_BRANCH-}" ]]; then
+		echo "${BASE_BRANCH}"
+		return 0
+	fi
+
+	local ref branch
+	ref=$(git symbolic-ref HEAD 2>/dev/null) || {
+		echo 'Not on a branch' >&2
+		return 1
+	}
+	branch="${ref#refs/heads/}"
+	if [[ "${ref}" == "${branch}" ]]; then
+		echo "Not on a branch: ${ref}" >&2
+		return 1
+	fi
+
+	if [[ "${branch}" != branch/v* ]]; then
+		# If we're already on the branch cut for the release, try to
+		# determine the root branch name
+		local fbranch
+		fbranch=$(
+			git branch \
+				--list 'branch/v*' \
+				--contains "$(git merge-base --fork-point HEAD)" \
+				--format '%(refname:short)'
+		)
+		branch="${fbranch:-${branch}}" # Don't overwrite $branch with empty
+	fi
+	if ! [[ "${branch}" == branch/v* ]]; then
+		echo "Not on a release branch: ${branch}" >&2
+		return 1
+	fi
+
+	echo "${branch}"
+	return 0
+}
+
+get_last_version() {
+	# If BASE_TAG is set, just use that, otherwise figure out the last version
+	if [[ -n "${BASE_TAG-}" ]]; then
+		echo "${BASE_TAG}"
+		return 0
+	fi
+
+	cd "$(git rev-parse --show-toplevel 2>/dev/null)" || {
+		echo 'ugh. cant cd to repo root' >&2
+		return 1
+	}
+	local last_version
+	last_version=$(make -s print-version) || {
+		echo 'Cant get last released version' >&2
+		return 1
+	}
+
+	echo "v${last_version}"
+	return 0
+}
+
+list_prs() {
+	local branch="$1" from="$2" to="${3-}"
+	local merged_query
+	if [[ -z "${to}" ]]; then
+		merged_query="merged:>${from}"
+	else
+		merged_query="merged:${from}..${to}"
+	fi
+
+	local gh_query="base:${branch} ${merged_query} -label:no-changelog"
 
-COMMIT=$(git rev-list -n 1 v$BASE_TAG)
+	# shellcheck disable=SC2016 # We're not trying to expand in single quotes
+	jq_expr='
```

Comment:
Can you please explain what this thing is doing? Maybe even add an example input and output? I cannot imagine trying to fix or update this expression without any context.